core/src/main/java/com/amplifyframework/api/graphql/GraphQLResponse.java

```diff
@@ -110,24 +113,73 @@ public int hashCode() {
      * See https://graphql.github.io/graphql-spec/June2018/#sec-Response-Format
      */
     public static final class Error {
+        // Description of the error
         private final String message;
 
+        // list of locations describing the syntax element
+        private final List<GraphQLLocation> locations;
+
+        // Details the key path of the response field with error.
+        private final List<GraphQLPathSegment> path;
+
+        // Additional error map, reserved for implementors to use however they see fit.
+        private final Map<String, Object> extensions;
+
         /**
          * Constructs error response in accordance with GraphQL specs.
          * @param message error message
+         * @param locations list of locations describing the syntax element
+         * @param path The key path of the response field with error.
+         * @param extensions additional error map, reserved for implementors.
          */
-        public Error(@NonNull String message) {
+        public Error(@NonNull String message, List<GraphQLLocation> locations, List<GraphQLPathSegment> path,
+                     Map<String, Object> extensions) {
             this.message = Objects.requireNonNull(message);
+            this.locations = locations;
+            this.path = path;
+            this.extensions = extensions;
         }
 
         /**
          * Gets the error message.
+         *
          * @return error message
          */
         public String getMessage() {
             return message;
         }
 
+        /**
+         * Gets the list of locations where each location describes the beginning of an associated
+         * syntax element.
+         *
+         * @return locations
+         */
+        public List<GraphQLLocation> getLocations() {
+            return locations;
```

Comment:
Let's make the list properties immutable. Luckily, we have nice `Immutable` utility class to help with that :)

```
return Immutable.of(locations);
```

Reply:
Also might be helpful to mark the return type as `@Nullable`, along with the other properties.

Reply:
Added!